oap-server/microbench/src/main/java/org/apache/skywalking/oap/server/microbench/base/AbstractMicrobenchmark.java

```diff
@@ -0,0 +1,116 @@
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * All JMH tests need to extend this class to make it easier for you to complete JMHTest, you can also choose to
+ * customize runtime conditions (Measurement, Fork, Warmup, etc.)
+ * <p>
+ * You can run any of the JMH tests as a normal UT, or you can package it and get all the reported results via `java
+ * -jar benchmark.jar`, or get the results of a particular Test via `java -jar /benchmarks.jar exampleClassName`.
+ */
+@Warmup(iterations = AbstractMicrobenchmark.DEFAULT_WARMUP_ITERATIONS)
+@Measurement(iterations = AbstractMicrobenchmark.DEFAULT_MEASURE_ITERATIONS)
+@Fork(AbstractMicrobenchmark.DEFAULT_FORKS)
+@State(Scope.Thread)
+@Slf4j
+public abstract class AbstractMicrobenchmark {
+    static final int DEFAULT_WARMUP_ITERATIONS = 10;
+
+    static final int DEFAULT_MEASURE_ITERATIONS = 10;
+
+    static final int DEFAULT_FORKS = 2;
+
+    public static class JmhThreadExecutor extends ThreadPoolExecutor {
+        public JmhThreadExecutor(int size, String name) {
+            super(size, size, 10, TimeUnit.SECONDS, new LinkedBlockingQueue<>(), Executors.defaultThreadFactory());
+        }
+    }
+
+    private ChainedOptionsBuilder newOptionsBuilder() {
+
+        String className = getClass().getSimpleName();
+
+        ChainedOptionsBuilder optBuilder = new OptionsBuilder()
+                // set benchmark class name
+                .include(".*" + className + ".*")
+                // add GC profiler
+                .addProfiler(GCProfiler.class)
+                //set jvm args
+                .jvmArgsAppend("-Xmx512m", "-Xms512m", "-XX:MaxDirectMemorySize=512m",
+                        "-XX:BiasedLockingStartupDelay=0",
+                        "-Djmh.executor=CUSTOM",
+                        "-Djmh.executor.class=org.apache.skywalking.oap.server.microbench.base.AbstractMicrobenchmark$JmhThreadExecutor"
+                );
+
+        String output = getReportDir();
+        if (output != null) {
+            boolean writeFileStatus;
+            String filePath = getReportDir() + className + ".json";
+            File file = new File(filePath);
+
+            if (file.exists()) {
+                writeFileStatus = file.delete();
+            } else {
+                writeFileStatus = file.getParentFile().mkdirs();
+                try {
+                    writeFileStatus = file.createNewFile();
+                } catch (IOException e) {
+                    log.warn("jmh test create file error", e);
+                }
+            }
+            if (writeFileStatus) {
+                optBuilder.resultFormat(ResultFormatType.JSON)
+                        .result(filePath);
+            }
+        }
+        return optBuilder;
+    }
+
+    @Test
+    public void run() throws Exception {
+        new Runner(newOptionsBuilder().build()).run();
+    }
+
+    private static String getReportDir() {
+        return System.getProperty("perfReportDir");
```

Comment:
This is not documented.

Reply:
Thank you for your reminder, I have explained it in the document, thank you again

Reply:
Sorry to find that this is a built-in feature (`-rff`) of JMH so we'd better not to replicate the feature.

```
java -jar oap-server/microbench/target/benchmarks.jar -rf json -rff /tmp/result.json AtomicRangeIntegerBenchmark
```

Reply:
This parameter needs to be added if the user uses IDE to run it.